src/pebble.py

```diff
@@ -401,6 +419,11 @@ def reconcile(  # noqa: C901 pylint: disable=too-many-branches,too-many-statemen
             ignore_order=True,
             ignore_string_case=True,
         )
+
+        restart_mas(container, rendered_mas_configuration)
```

Comment:
Considering that MAS is now a requirement for Synapse (right?), I think this could be done as the first step, before anything related to Synapse.